sdks/java/core/src/main/java/org/apache/beam/sdk/io/GenerateSequence.java

```diff
@@ -98,6 +101,24 @@
     abstract GenerateSequence build();
   }
 
+  public static GenerateSequence fromExternal(ExternalTransforms.GenerateSequencePayload payload) {
```

Comment:
As mentioned in the doc, I'm bit worried about this extra work on the transform author to make a given transform cross-language compatible. I think we should aim for a model where the transform author simply describe the interface for initiating the transform using a schema. Extra work for building the transform should be generated and performed by the framework.

This is probably OK as a first version though.

Reply:
I share your concerns. Building IO and exposing it to the SDKs should be as simple as possible. Perhaps we keep the Proto-based approach for advanced users, but there should be a easy to use schema-based approach which automatically plugs in the transform and generates Proto messages.